ner_v1/detectors/textual/name/name_detection.py

```diff
@@ -243,6 +256,63 @@ def detect_hindi_name(self):
 
         return entity_value, original_text
 
+    def replace_free_text_detection_text(self, free_text_detection_results, text):
+        """
+        Replace detected names from the text according to replace_detected_text.
+        Separate method for replacing free_text_detection_results because it these results are not at token level.
+        For example -
+            text = "my name is yash doshi"
+            free_text_detection_results = ["yash doshi"]
+            while, text_detection_original_texts = ["yash", "doshi"]
+
+
+        Args:
+            free_text_detection_results(list): list containing free_text_entity_results
+            text(str): original to run detection on
+
+        Returns:
+            replaced_text(str): text with marked tokens
+
+        Example:
+            >> text = "my name is yash doshi"
+            >> free_text_detection_results = ["yash doshi"]
+            >> replace_free_text_detection_text(free_text_detection_results, text)
+            'my name is _yash_ _doshi_'
+
+        """
+        if self.language == ENGLISH_LANG:
+            replaced_original_text_tokens = nltk_tokenizer.tokenize(text.lower())
+            replaced_text_tokens = []
+            for index, token in enumerate(replaced_original_text_tokens):
+                # Fix to handle tokenizer error for tokens with trailing `.`. For eg.
+                # >> text = "my name is v.k. singh"
+                # >> tokens = tokenize(text)
+                # >> tokens
+                #    ["my", "name", "is", "v.k", ".", "singh"]
+                # this extra `.` token causes problem while training.
+                if token == "." and 0 < index < len(replaced_original_text_tokens) - 1 \
+                        and replaced_text_tokens[-1] + "." in text.lower():
+                    replaced_text_tokens[-1] = replaced_text_tokens[-1] + "."
+                else:
+                    # fix to handle examples like `miami,21st street` where tokenizer gives ["miami,21st", "street"].
```

Comment:
Did you mean ["miami,", "21st", "street"]

Reply:
Done